### ISSUE TITLE
Communication problem between recreated users

### ISSUE THREAD

**Issue by juanrhino** [RocketChat/Rocket.Chat#6190]
Rocket.Chat Version: 0.5.2
Running Instances: 1
DB Replicaset OpLog: No
Node Version: 4.8.0

When you delete a user and create it again some other users cannot chat with that user it says that the channel doesn't exists. If we create a channel they can communicate but never directly

**Comment by ledainam** [comment]
i have same bug 
they always said room isn't exist
![bug](https://cloud.githubusercontent.com/assets/25676833/23686274/15f744e6-03db-11e7-98a5-91ac5e06a9bf.png)

**Comment by 0x6c** [comment]
Same for me +1

**Comment by 0x6c** [comment]
It's becomes very important, as linking with ad and user recreation is usual operation.

**Comment by localguru** [comment]
I can confirm on 0.57.0, see attached image. 

I removed the user "Testine Test" (left side in Firefox) and created it again (I use SAML, so with a new login the user is recreated automatically). On the right side (in Chrome) the user "Al Uminium" was in direct message DM with user "Testine Test" before. After recreation the user "Al Uminium" can't find user "Testine Test" anymore. Curiously user "Testine Test" can send DM to "Al Uminium", see pop up message. I restarted the node, that doesn't help. I had to remove use "Al Uminium" too, to get DM working between them again. 

I guess it has something to do with the internal unique ID and all other users which were in DM before still use the old ID.


![room_not_found](https://user-images.githubusercontent.com/8544984/28242400-d25ef040-69aa-11e7-8d18-952aab3abe45.jpg)

**Comment by TheReal1604** [comment]
I think this has something to do with the client side cache. You can easily reproduce this if you delete a channel and recreate it. You don't see messages first. But if you refresh your browser cache all is working as expected. Maybe @localguru can you try this?

**Comment by localguru** [comment]
@TheReal1604 yes, you are right! Cleaning the cache did it. Good to know. Thanks!

**Comment by TheReal1604** [comment]
@rodrigok Maybe you can take a look as you are the cache expert? 👍  😁
